ocs_ci/ocs/warp.py

```diff
@@ -0,0 +1,189 @@
+    built to conduct tests between WARP clients and object storage hosts.
+    WARP measures GET and PUT performance from multiple clients against a MinIO cluster.
+
+    """
+
+    def __init__(self):
+        """
+        Initializer to create warp pod to running warp benchmark
+
+        """
+        self.pod_dic_path = constants.WARP_YAML
+        self.namespace = config.ENV_DATA["cluster_namespace"]
+        self.ocp_obj = OCP(namespace=self.namespace)
+        self.bucket_name = None
+        self.warp_cr = templating.load_yaml(constants.WARP_OBJ_YAML)
+        self.host = self.warp_cr["host"]
+        self.duration = self.warp_cr["duration"]
+        self.concurrent = self.warp_cr["concurrent"]
+        self.objects = self.warp_cr["objects"]
+        self.obj_size = self.warp_cr["obj.size"]
+        self.get_distrib = self.warp_cr["get-distrib"]
+        self.put_distrib = self.warp_cr["put-distrib"]
+        self.delete_distrib = self.warp_cr["delete-distrib"]
+        self.stat_distrib = self.warp_cr["stat-distrib"]
+        self.warp_bin_dir = self.warp_cr["warp_bin_dir"]
+        self.output_file = "output.csv"
+        self.warp_dir = mkdtemp(prefix="warp-")
+
+    def create_resource_warp(self):
+        """
+        Create resource for Warp S3 benchmark:
+            * Create service account
+            * Create PVC
+            * Create warp pod for running workload
+
+        """
+
+        # Create service account
+        self.sa_name = helpers.create_serviceaccount(self.namespace)
+        self.sa_name = self.sa_name.name
+        helpers.add_scc_policy(sa_name=self.sa_name, namespace=self.namespace)
+
+        # Create test pvc+pod
+        log.info(f"Create Warp pod to generate S3 workload in {self.namespace}")
+        pvc_size = "50Gi"
+        self.pod_name = "warp-pod"
+        self.pvc_obj = helpers.create_pvc(
+            sc_name=constants.DEFAULT_STORAGECLASS_RBD,
+            namespace=self.namespace,
+            size=pvc_size,
+        )
+        self.pod_obj = helpers.create_pod(
+            constants.CEPHBLOCKPOOL,
+            namespace=self.namespace,
+            pod_name=self.pod_name,
+            pvc_name=self.pvc_obj.name,
+            sa_name=self.sa_name,
+            pod_dict_path=self.pod_dic_path,
+            dc_deployment=True,
+            deploy_pod_status=constants.STATUS_COMPLETED,
+        )
+
+    def install_warp(self, timeout=1800):
+        """
+        Install minio warp S3 benchmark:
+        https://github.com/minio/warp
+
+        """
+        # Install warp
+        log.info(f"Installing warp S3 benchmark on testing pod {self.pod_obj.name}")
+        warp_path = "https://github.com/minio/warp/releases/download/v0.5.5/warp_0.5.5_Linux_x86_64.tar.gz"
+        self.pod_obj.exec_cmd_on_pod(f"wget {warp_path}", timeout=timeout)
+        self.pod_obj.exec_cmd_on_pod(
+            "tar -xzvf warp_0.5.5_Linux_x86_64.tar.gz", timeout=timeout
+        )
```

Comment:
Would it be good to save this .tgz file at our quay repo ?
so if it will be deleted from the source we will still have it.
or better, crate a pre-installed container image and use it - it will save time (and / or issues) of downloading & installing the benchmark.

Reply:
Fixed.  Added warp binary in quay.io image with v0.6.2 version and remove install function in `warp.py`.